Sources/Sentry/SentryProfiler.mm

```diff
@@ -415,26 +462,85 @@ + (void)useFramesTracker:(SentryFramesTracker *)framesTracker
+            [array.lastObject[@"elapsed_since_start_ns"] longLongValue]
+            + _gCurrentProfiler->_startTimestamp;
+        const auto firstSampleRelativeToTransactionStart
+            = firstSampleAbsoluteTime - transaction.startSystemTime;
+        const auto lastSampleRelativeToTransactionStart
+            = lastSampleAbsoluteTime - transaction.startSystemTime;
+        SENTRY_LOG_DEBUG(@"[slice start] Could not find any sample taken during the transaction "
+                         @"(first sample taken at: %llu; last: %llu; transaction start: %llu; end: "
+                         @"%llu; first sample relative to transaction start: %lld; last: %lld).",
+            firstSampleAbsoluteTime, lastSampleAbsoluteTime, transaction.startSystemTime,
+            transaction.endSystemTime, firstSampleRelativeToTransactionStart,
+            lastSampleRelativeToTransactionStart);
+        return nil;
     } else {
-        SENTRY_LOG_DEBUG(@"Profiler %@ is waiting for more spans to complete: %@.", profiler,
-            profiler->_spansInFlight);
+        SENTRY_LOG_DEBUG(@"Found first slice sample at index %lu", firstIndex);
+    }
+
+    const auto lastIndex =
+        [array indexOfObjectWithOptions:NSEnumerationReverse
+                            passingTest:^BOOL(NSDictionary<NSString *, id> *_Nonnull sample,
+                                NSUInteger idx, BOOL *_Nonnull stop) {
+                                const auto absoluteSampleTime =
+                                    [sample[@"elapsed_since_start_ns"] longLongValue]
+                                    + _gCurrentProfiler->_startTimestamp;
+                                *stop = absoluteSampleTime <= transaction.endSystemTime;
+                                return *stop;
+                            }];
+
+    if (lastIndex == NSNotFound) {
+        const auto firstSampleAbsoluteTime = [array[0][@"elapsed_since_start_ns"] longLongValue]
+            + _gCurrentProfiler->_startTimestamp;
+        const auto lastSampleAbsoluteTime =
+            [array.lastObject[@"elapsed_since_start_ns"] longLongValue]
+            + _gCurrentProfiler->_startTimestamp;
+        const auto firstSampleRelativeToTransactionStart
+            = firstSampleAbsoluteTime - transaction.startSystemTime;
+        const auto lastSampleRelativeToTransactionStart
+            = lastSampleAbsoluteTime - transaction.startSystemTime;
+        SENTRY_LOG_DEBUG(
+            @"[slice end] Could not find any other sample taken during the transaction (first "
+            @"sample taken at: %llu; last: %llu; transaction start: %llu; end: %llu; first sample "
+            @"relative to transaction start: %lld; last: %lld).",
+            firstSampleAbsoluteTime, lastSampleAbsoluteTime, transaction.startSystemTime,
+            transaction.endSystemTime, firstSampleRelativeToTransactionStart,
+            lastSampleRelativeToTransactionStart);
+        return nil;
```

Comment:
`l`: This code block looks almost a duplicate of https://github.com/getsentry/sentry-cocoa/blob/56141bd6dfb7d4908a40df537f7f56beb42a83cd/Sources/Sentry/SentryProfiler.mm#L475-L491

It looks like we could extract a method to reduce duplicated code.